src/content/docs/identityserver/upgrades/identityserver4-to-duende-identityserver-v7.mdx

```diff
@@ -0,0 +1,513 @@
+```
+
+Replace all `IdentityModel` namespace usages with `Duende.IdentityModel`:
+
+```diff lang="csharp" title="*.cs"
+- using IdentityModel;
++ using Duende.IdentityModel;
+```
+
+If you are using fully-qualified names in your code, those will need to be updated as well.
+
+### Step 4: Remove AddDeveloperSigningCredential <Badge text="Optional" variant="note" size="small" />
+
+In your application startup code, typically found in the `ConfigureServices` method in `Startup.cs`, consider removing `AddDeveloperSigningCredential`.
+
+You can use Duende IdentityServer's built-in [manual or automatic key management](../../fundamentals/key-management) instead.
+
+### Step 5: Update Database Schema
+
+Whether you are using a [database](../../data) or a [custom store implementation](../..//reference/stores) for your configuration and operational data, you'll need to make some changes.
+The exact steps involved in updating your data store will depend on your implementation details.
+
+- Create a new `Keys` table for the automatic key management feature in the operational database.
+- Create a new `RequireResourceIndicator` boolean column on the `ApiResources` table in the configuration database.
+- Create a new index on the `ConsumedTime` column in the `PersistedGrants` table ([more details](https://github.com/DuendeSoftware/products/pull/84)).
+- Create a new table called `IdentityProviders` for storing the OIDC provider details ([more details](https://github.com/DuendeSoftware/products/pull/188)).
+- Add missing columns for created, updated, etc. to EF entities ([more details](https://github.com/DuendeSoftware/products/pull/356)).
+- Add unique constraints to EF tables where duplicate records not allowed ([more details](https://github.com/DuendeSoftware/products/pull/355)).
+- The server-side sessions feature requires a new table ([more details](https://github.com/DuendeSoftware/products/pull/743)).
+- The session coordination feature adds a column to the `Clients` table ([more details](https://github.com/DuendeSoftware/products/pull/820)).
+- Improve primary keys on the persisted grants table ([more details](https://github.com/DuendeSoftware/products/pull/793)).
+- Add new properties to the [`Duende.IdentityServer.Models.Client` model](/identityserver/reference/models/client):
+
+  - `InitiateLoginUri` is a nullable string used for Third Party Initiated Login.
+  - `RequireDPoP` is a non-nullable boolean flag that controls if a client is required to use [DPoP](../../tokens/pop).
+  - `DPoPValidationMode` is a non-nullable column that controls the DPoP validation mechanism. Existing clients that are not using DPoP can set its value to `0`.
+  - `DPoPClockSkew` is a non-nullable timespan that controls how much clock skew is allowed for a particular DPoP client. Existing clients that are not using DPoP can set its value to a timespan of length ``0.
+
+- The `ServerSideSession` entity now uses a 64-bit long as its primary key.
```

Comment:
This is not a relevant change if they go directly from IdSrv4 to IdSrv7. It was only in v5 (if I remember right) that this was an int.